lib/minfraud/transaction.rb

```diff
@@ -65,46 +48,18 @@ def attributes
       Hash[attrs]
     end
 
-    # Uses the requested_type set on the instance, or if not present, the requested_type set during configuration
-    # @return [String, nil] requested type
-    def requested_type
-      @requested_type or Minfraud.requested_type
-    end
-
-    # Sends transaction to MaxMind in order to get risk data on it.
-    # Caches response object in @response.
-    # @return [Response]
-    def response
-      @response ||= Request.get(self)
-    end
-
-    def timeout=(timeout)
-      raise ArgumentError, "Timeout value must be Numeric" unless timeout.is_a?(Numeric)
-      @timeout = timeout
-    end
-
-    def open_timeout=(timeout)
-      raise ArgumentError, "Open timeout value must be Numeric" unless timeout.is_a?(Numeric)
-      @open_timeout = timeout
-    end
-
-    def read_timeout=(timeout)
-      raise ArgumentError, "Read timeout value must be Numeric" unless timeout.is_a?(Numeric)
-      @read_timeout = timeout
-    end
-
     private
 
     # Ensures the required attributes are present
     # @return [Boolean]
     def has_required_attributes?
-      ip && txn_id
+      ip && txn_id && license_key
     end
 
     # Validates the types of the attributes
     # @return [nil, TransactionError]
     def validate_attributes
-      [:ip, :city, :state, :postal, :country, :txn_id].each { |s| validate_string(s) }
+      [:ip, :city, :state, :postal, :country, :txn_id, :license_key].each { |s| validate_string(s) }
     end
```

Comment:
@dharkael you mentioned cleaning up the transaction validation, is this what you were talking about? Currently these attributes are only checked that they are strings, but the [minFraud API docs](https://dev.maxmind.com/minfraud/minfraud-legacy/) give more information on what it actually expects